frappe/app.py

```diff
@@ -237,8 +243,26 @@ def process_response(response):
 	if not response:
 		return
 
-	# set cookies
-	if hasattr(frappe.local, "cookie_manager"):
+	# cache control
+	# read: https://simonhearne.com/2022/caching-header-best-practices/
+	if frappe.local.response.can_cache:
+		response.headers.extend(
+			{
+				# default: 5m (proxy), 5m (client), 3h (allow stale resources for this long if upstream is down)
+				"Cache-Control": "public,s-maxage=300,max-age=300,stale-while-revalidate=10800",
```

Comment:
Aren't a lot of endpoints user-specific? `public` is possibly bad idea here. 2/4 endpoints here are user-specific: https://github.com/frappe/frappe/pull/26753 

Proxy caching should just be skipped, it doesn't help nearly as much as client-side caching in our case. Only good use case I can think of is static pages.